OWML.ModHelper/ModMergedConfig.cs

```diff
@@ -60,120 +57,85 @@ public void SaveToStorage()
-            }
-            else if (_defaultConfig != null)
-            {
-                var settingsChanged = !MakeConfigConsistentWithDefault();
-                if (settingsChanged)
-                {
-                    ModConsole.OwmlConsole.WriteLine($"Warning - Settings mod {_manifest.UniqueName} changed", MessageType.Warning);
-                }
-            }
-            SaveToStorage();
-        }
-
-        private bool MakeConfigConsistentWithDefault()
-        {
-            var wasCompatible = true;
-            var toRemove = _userConfig.Settings.Keys.Except(_defaultConfig.Settings.Keys).ToList();
-            toRemove.ForEach(key => _userConfig.Settings.Remove(key));
-
-            var keysCopy = _userConfig.Settings.Keys.ToList();
-            foreach (var key in keysCopy)
-            {
-                if (!IsSettingSameType(_userConfig.Settings[key], _defaultConfig.Settings[key]))
-                {
-                    wasCompatible = TryUpdate(key, _userConfig.Settings[key], _defaultConfig.Settings[key]) && wasCompatible;
-                }
-                else if (_defaultConfig.Settings[key] is JObject objectValue && objectValue["type"].ToString() == "selector")
-                {
-                    wasCompatible = UpdateSelector(key, _userConfig.Settings[key], objectValue) && wasCompatible;
-                }
+                return jObject["value"].ToObject(typeof(object));
             }
-
-            AddMissingDefaults(_defaultConfig);
-            return wasCompatible;
+            return outerValue;
         }
 
-        private bool UpdateSelector(string key, object userSetting, JObject modSetting)
+        private void SetInnerValue(Dictionary<string, object> settings, string key, object value)
         {
-            var options = modSetting["options"].ToObject<List<string>>();
-            var userString = userSetting is JObject objectValue ? (string)objectValue["value"] : Convert.ToString(userSetting);
-            _userConfig.Settings[key] = modSetting;
-            var isInOptions = options.Contains(userString);
-            if (isInOptions)
+            if (settings[key] is JObject jObject)
             {
-                _userConfig.SetSettingsValue(key, userString);
+                jObject["value"] = JToken.FromObject(value);
+                return;
             }
-            return isInOptions;
+            settings[key] = value;
         }
 
-        private void AddMissingDefaults(IModConfig defaultConfig)
+        private bool IsNumber(object value)
         {
-            var missingSettings = defaultConfig.Settings.Where(s => !_userConfig.Settings.ContainsKey(s.Key)).ToList();
-            missingSettings.ForEach(setting => _userConfig.Settings.Add(setting.Key, setting.Value));
+            return value is int
+                || value is long
+                || value is float
+                || value is double
+                || value is decimal
+                || value is ulong;
```

Comment:
I feel like this could be much simpler, but I don't seem to understand how Json.Net decides to which type it will convert a JSON property. From my tests it seems like our numbers are always either long or float, but I have no idea if it will suddenly decide to use some other number type, so I used all the types that are listed in Json.Net's documentation.

A lot of this stuff could be heavily simplified if we had our own serializer, meaning we would always know exactly what types we're dealing with, and we wouldn't need to mess directly with `object`, `JObject`, etc, and convert them back and forth so much.